Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderAssetBuilder.cpp

```diff
@@ -59,13 +59,99 @@ namespace AZ
         static constexpr char ShaderAssetBuilderName[] = "ShaderAssetBuilder";
         static constexpr uint32_t ShaderAssetBuildTimestampParam = 0;
 
+        static AZ::Outcome<AZStd::vector<AZStd::string>, AZStd::string> GetListOfIncludedFiles(AZStd::string_view sourceFilePath)
+        {
+            AZ::IO::FileIOStream stream(sourceFilePath.data(), AZ::IO::OpenMode::ModeRead);
+            if (!stream.IsOpen())
+            {
+                return AZ::Failure(AZStd::string::format("\"%s\" source file could not be opened.", sourceFilePath.data()));
+            }
+
+            if (!stream.CanRead())
+            {
+                return AZ::Failure(AZStd::string::format("\"%s\" source file could not be read.", sourceFilePath.data()));
+            }
+
+            // Cache the path of the folder where @sourceFilePath.
+            AZStd::string sourceFileFolderPath;
+            {
+                AZStd::string drive;
+                AzFramework::StringFunc::Path::Split(sourceFilePath.data(), &drive, &sourceFileFolderPath);
+                if (!drive.empty())
+                {
+                    AzFramework::StringFunc::Path::Join(drive.c_str(), sourceFileFolderPath.c_str(), sourceFileFolderPath);
+                }
+            }
+
+            AZStd::string hayStack;
+            hayStack.resize(stream.GetLength());
```

Comment:
You can probably use resize_no_construct here to save more time.

Reply:
Will do.